src/rpc/rawtransaction.cpp

```diff
@@ -970,54 +970,52 @@ UniValue createrawtransaction(const JSONRPCRequest& request)
 
                     if (keys.size() == 0)
                         throw JSONRPCError(RPC_INVALID_PARAMETER, std::string(
                                 "Invalid parameter, the format must follow { \"transferwithmessage\": {\"asset_name\": amount, \"message\": messagehash, \"expire_time\": utc_time} }"));
 
-                    UniValue asset_quantity;
                     std::string asset_name = keys[0];
 
-                    if (!IsAssetNameValid(asset_name)) {
+                    if (!IsAssetNameValid(asset_name)) 
                         throw JSONRPCError(RPC_INVALID_PARAMETER,
                                            "Invalid parameter, missing valid asset name to transferwithmessage");
 
-                        const UniValue &asset_quantity = find_value(transferData, asset_name);
-                        if (!asset_quantity.isNum())
-                            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, missing or invalid quantity");
+                    const UniValue &asset_quantity = find_value(transferData, asset_name);
+                    if (!asset_quantity.isNum())
+                        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, missing or invalid quantity");
 
-                        const UniValue &message = find_value(transferData, "message");
-                        if (!message.isStr())
-                            throw JSONRPCError(RPC_INVALID_PARAMETER,
-                                               "Invalid parameter, missing reissue data for key: message");
+                    const UniValue &message = find_value(transferData, "message");
+                    if (!message.isStr())
+                        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                                           "Invalid parameter, missing reissue data for key: message");
 
-                        const UniValue &expire_time = find_value(transferData, "expire_time");
-                        if (!expire_time.isNum())
-                            throw JSONRPCError(RPC_INVALID_PARAMETER,
-                                               "Invalid parameter, missing reissue data for key: expire_time");
+                    const UniValue &expire_time = find_value(transferData, "expire_time");
+                    if (!expire_time.isNum())
+                        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                                           "Invalid parameter, missing reissue data for key: expire_time");
 
-                        CAmount nAmount = AmountFromValue(asset_quantity);
+                    CAmount nAmount = AmountFromValue(asset_quantity);
 
-                        // Create a new transfer
-                        CAssetTransfer transfer(asset_name, nAmount, DecodeAssetData(message.get_str()),
+                    // Create a new transfer
+                    CAssetTransfer transfer(asset_name, nAmount, DecodeAssetData(message.get_str()),
```

Comment:
There are no change addresses here so you can not dictate where change will go. 

There should be an asset change value and a rvn change value to conform to the standard transfer rpc format.